src/components/VHeader/VHeader.vue

```diff
@@ -0,0 +1,281 @@
+
+    const { isVisible: isFilterVisible } = useFilterSidebarVisibility()
+
+    /**
+     * Set the active mobile menu view to the 'filters'
+     * if the filter sidebar has been toggled open.
+     */
+    watch([isFilterVisible], ([isFilterVisible]) => {
+      openMenu.value = isFilterVisible ? menus.FILTERS : null
+    })
+
+    /**
+     * @type {import('@nuxtjs/composition-api').Ref<null|'filters'|'content-switcher'>}
+     */
+    const openMenu = ref(null)
+    const isMenuOpen = computed(() => openMenu.value !== null)
+
+    /**
+     * @param {'filters'|'content-switcher'} menuName
+     */
+    const openMenuModal = (menuName) => {
+      if (openMenu.value !== null) {
+        close()
+      }
+      openMenu.value = menuName
+    }
+    const close = () => {
+      openMenu.value = null
+    }
+    /**
+     * Only returns true on search route (?) on mobile screen.
+     * Necessary to show 'close' button in the header.
+     * @type {import('@nuxtjs/composition-api').ComputedRef<boolean>} */
+    const showCloseButton = computed(() => {
+      return !isMinScreenMd.value && openMenu.value !== null
+    })
+
+    /**  @type {import('@nuxtjs/composition-api').ComputedRef<boolean>} */
+    const isFetching = computed(() => {
+      return store.getters['media/fetchState'].isFetching
+    })
+
+    /**
+     * Return a text representation of the result count.
+     * @param {number} count
+     * @returns {string}
+     */
+    const mediaCount = (count) => {
+      const countKey =
+        count === 0 ? 'noResult' : count >= 10000 ? 'more' : 'result'
+      const i18nKey = i18nKeys[store.state.search.query.mediaType][countKey]
+      const localeCount = count.toLocaleString(i18n.locale)
+      return i18n.tc(i18nKey, count, { localeCount })
+    }
+
+    /** @type {import('@nuxtjs/composition-api').ComputedRef<number>} */
+    const resultsCount = computed(() => store.getters['media/results'].count)
+
+    /**
+     * Status is hidden below the medium breakpoint.
+     * It shows Loading... or Number of results on bigger screens.
+     * @returns {string}
+     */
+    const setInitialStatus = () => {
+      if (!isMinScreenMd.value) return ''
+      if (isFetching.value) return i18n.t('header.loading')
+      return store.state.search.query.q === ''
+        ? ''
+        : mediaCount(resultsCount.value)
+    }
+
+    /** @type {import('@nuxtjs/composition-api').Ref<string>} */
+    const searchStatus = ref(setInitialStatus())
+
+    watchEffect(() => {
+      if (isMinScreenMd.value) {
+        searchStatus.value = isFetching.value
+          ? i18n.t('header.loading')
+          : mediaCount(resultsCount.value)
+      } else {
+        searchStatus.value = ''
+      }
+    })
+
+    const localSearchTerm = ref(store.state.search.query.q)
+    const searchTerm = computed({
+      get: () => localSearchTerm.value,
+      set: async (value) => {
+        localSearchTerm.value = value
+      },
+    })
```

Comment:
This is used for `v-model`, as it has to have a getter and a setter to be updatable. `localSearchTerm` as a ref needs to use a different syntax to set the value (`localSearchTerm.value = val` instead of `localSearchTerm = val`), so it doesn't work here.